prefect_snowflake/credentials.py

```diff
@@ -119,3 +144,91 @@ def _validate_okta_kwargs(cls, values):
                 "`okta_endpoint` must be provided"
             )
         return values
+
+    @validator("private_key")
+    def _validate_private_key(cls, private_key):
+        """
+        Ensure a private_key looks like a PEM format certificate.
+        """
+
+        if private_key is None:
+            return None
+
+        assert isinstance(private_key, SecretBytes)
+
+        pk = cls._decode_secret(private_key)
+
+        return None if pk is None else SecretBytes(cls._compose_pem(pk))
+
+    def resolve_private_key(self) -> Optional[bytes]:
+        """
+        Converts a PEM encoded private key into a DER binary key
+
+        Returns:
+            DER binary key if private_key has been provided
+                    otherwise returns None
+
+        Raises:
+            InvalidPemFormat: if `private_key` is not in PEM format
+        """
+
+        private_key = self._decode_secret(self.private_key)
+
+        if private_key is None:
+            return None
+
+        return load_pem_private_key(
+            data=private_key,
+            password=self._decode_secret(self.password),
+            backend=default_backend(),
+        ).private_bytes(
+            encoding=Encoding.DER,
+            format=PrivateFormat.PKCS8,
+            encryption_algorithm=NoEncryption(),
+        )
+
+    @staticmethod
+    def _decode_secret(secret: Union[SecretStr, SecretBytes]) -> Optional[bytes]:
+        """
+        Decode the provided secret into bytes. If the secret is not a
+        string or bytes, or it is whitespace, then return None.
+
+        Args:
+            secret: The value to decode.
+
+        Returns:
+            The decoded secret as bytes.
+
+        """
+        if isinstance(secret, (SecretBytes, SecretStr)):
+            secret = secret.get_secret_value()
+
+        if not isinstance(secret, (bytes, str)) or len(secret) == 0 or secret.isspace():
+            return None
+
+        return secret if isinstance(secret, bytes) else secret.encode()
+
+    @staticmethod
+    def _compose_pem(private_key: bytes) -> bytes:
+        """Validate structure of PEM certificate.
+
+        The original key passed from Prefect is sometimes malformed.
+        This function recomposes the key into a valid key that will
+        pass the serialization step when resolving the key to a DER.
+
+        Args:
+            private_key (str): A valid PEM format string
+
+        Returns:
+            Binary encoded certificate
+
+        Raises:
+            InvalidPemFormat: if `private_key` is an invalid format
+        """
```

Comment:
Can we double check whether this looks okay on docs by running `mkdocs serve`?

Reply:
I've revised this.